plugins/guests/solaris11/guest.rb

```diff
@@ -8,7 +8,7 @@ module VagrantPlugins
   module GuestSolaris11
     class Guest < Vagrant.plugin("2", :guest)
       def detect?(machine)
-        machine.communicate.test("uname -sr | grep 'SunOS.*5\\.11'")
```

Comment:
I am unsure what compatibility impacts this modification will create for other solaris guests. From first glance this seems like it may easily break guests that currently are in working order.

Reply:
If they are real Solaris 11 guests it will not make any impact because the information is also provided in `/etc/release` which show the information that it is a real Solaris 11 which somehow is not true with `uname -sr` especially for SmartOS, OmniOS and other distributions.

I made this change especially to be sure vagrant detect the SmartOS machine as `smartos` guest and not as `solaris11` guest. First of all to have the valid `smartos` guest features enabled the second was that some commands used in the `solaris11` guest network configuration do not work on SmartOS (or OmniOS).

Reply:
@drscream :+1: thanks for the explanation!